test/users.test.js

```diff
@@ -77,4 +79,52 @@ describe('Users', () => {
         .catch(err => done(err));
     });
   });
+
+  describe('POST /users/session', () => {
+    test('token should be returned', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getUserByEmailMock);
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .then(res => {
+          expect(res.body.data.token).toBeTruthy();
+          done();
+        })
+        .catch(err => done(err));
+    });
+
+    test('bad request should be returned when password does not match', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getUserByEmailMock);
+      signInInput.password = 'abc12345';
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(400)
```

Comment:
400 http status code means that the request was malformed. I think you should use 401 unauthorized when the login fails.

Reply:
ok